### ISSUE TITLE
WPF with PackageReference and referencing .NET Core project produces NuGet error on build

### ISSUE THREAD

**Issue by AArnott** [NuGet/Home#4652]
WPF stopped building reliably after I added my first PackageReference to it. 

```
git clone https://github.com/AArnott/MoneyMan.git
cd MoneyMan
git checkout bugrepro/nuget/4652
cd src
msbuild /t:restore,build
```

**Expected**

Successful build. I've seen this occasionally on this source version, suggesting that this is a non-deterministic failure.

**Actual**

> C:\Program Files (x86)\Microsoft Visual Studio\2017\d15rel\MSBuild\Microsoft\NuGet\15.0\Microsoft.NuGet.targets(197,5): error : The project.json is referencing the project 'C:\git\MoneyMan\obj\Nerdbank.MoneyManagement\Nerdbank.MoneyManagement.csproj', but an output path was not specified on an item in the ProjectReferencesCreatingPackages property. [C:\git\MoneyMan\src\MoneyMan.WPF\MoneyMan.WPF.csproj]

[Also as seen on Appveyor](https://ci.appveyor.com/project/AArnott/moneyman/build/0.1.4-alpha+gbb11765fb1)

Repro machine has: d15rel 26217.2

**Comment by emgarten** [comment]
This issue is tracked in https://github.com/NuGet/Home/issues/4532

@AArnott would you try the workaround listed in #4532 and see if it works for you?

**Comment by AArnott** [comment]
Thanks for the tip, @emgarten. The workaround appears to be incomplete. Although it avoids the error message I report above, it is replaced by several new ones from the WPF temporary project, which evidently doesn't contain the resolved package references that the original WPF project does:

> ViewModels\AccountsPanelViewModel.cs(13,11): error CS0246: The type or namespace name 'PCLCommandBase' could not be found (are you missing a using directive or an assembly reference?) [C:\git\MoneyMan\src\MoneyMan.WPF\rkqstjls.tmp_proj]
ViewModels\MainPageViewModel.cs(11,11): error CS0246: The type or namespace name 'PCLCommandBase' could not be found (are you missing a using directive or an assembly reference?)
[C:\git\MoneyMan\src\MoneyMan.WPF\rkqstjls.tmp_proj]
ViewModels\AccountsPanelViewModel.cs(15,43): error CS0246: The type or namespace name 'BindableBase' could not be found (are you missing a using directive or an assembly reference?) [C:\git\MoneyMan\src\MoneyMan.WPF\rkqstjls.tmp_proj]
ViewModels\MainPageViewModel.cs(13,38): error CS0246: The type or namespace name 'BindableBase' could not be found (are you missing a using directive or an assembly reference?) [C:\git\MoneyMan\src\MoneyMan.WPF\rkqstjls.tmp_proj]

These errors are consistent with what I would expect if this PackageReference were missing from the original WPF project:

```xml
    <PackageReference Include="PCLCommandBase" Version="0.3.5" />
```

Of course, it _is_ there, but the tmp_proj probably isn't resolving it.

**Comment by rrelyea** [comment]
Dupe of #4532 according to @zhili1208

**Comment by rrelyea** [comment]
@AArnott - can you confirm that this is fixed nowadays? the duplicate bug was fixed in 4.1.

**Comment by AArnott** [comment]
@rrelyea Happy to test. How do I use 4.1? What vehicle does it ship in?

**Comment by zhili1208** [comment]
just use VS 2017 15.3 to restore and build

**Comment by AArnott** [comment]
Yes, it appears to be fixed in 15.3.0. Thanks.

**Comment by HEskandari** [comment]
There is still an issue with PackageReference that contains source files (imported with PrivateAssets="All"). The source files are not picked up at compile time. This is specific to `Microsoft.NET.Sdk.WindowsDesktop` SDK projects.
